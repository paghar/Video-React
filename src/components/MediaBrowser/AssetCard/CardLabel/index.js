//@flow
import React from 'react';
import { LabelWrapper, LabelCaption } from './index.style';
import { Generals } from 'theme/colors';
import { Display6 } from 'theme/fonts';

type Props = {
  labelCaption: string,
  labelColor?: string,
  className?: string,
};

const CardLabel = ({ className, labelCaption, labelColor }: Props) => {
  return (
    <LabelWrapper className={className} haveColor={!!labelColor}>
      <LabelCaption
        font={Display6}
        color={Generals.viii}
        hoverable
        labelColor={labelColor}
      >
        {labelCaption}
      </LabelCaption>
    </LabelWrapper>
  );
};

export default CardLabel;
