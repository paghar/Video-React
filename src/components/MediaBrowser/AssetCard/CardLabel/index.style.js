//@flow
import styled from 'styled-components';
import { Generals } from 'theme/colors';
import { hexToRgb } from 'helpers';
import Typography from 'components/shared/Typography';

export const LabelWrapper = styled.div`
  height: 27px;
  background-color: ${({ ...props }) => hexToRgb(Generals.ii(props), 0.7)};
  border-radius: 5px;
  text-align: center;
  padding-left: ${({ haveColor }) => (haveColor ? '19' : '8')}px;
  padding-top: 3px;
  padding-right: ${({ haveColor }) => (haveColor ? '10' : '7')}px;
`;

export const LabelCaption = styled(Typography)`
  &:before {
    ${({ labelColor }) =>
      labelColor
        ? `
    content: ' ';
    width: 7px;
    height: 7px;
    background-color: ${labelColor};
    border-radius: 100%;
    position: absolute;
    top: 10px
    left: 8px;
    `
        : ''}
  }
`;
