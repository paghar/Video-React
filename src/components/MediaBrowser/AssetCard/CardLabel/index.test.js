//@flow
import React from 'react';
import renderer from 'react-test-renderer';
import CardLabel from '.';

describe('<AssetCardLabel />', () => {
  it('AssetCardLabel Snapshot test', () => {
    const tree = renderer
      .create(<CardLabel labelCaption="In Progress" labelColor="green" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('AssetCardLabel Snapshot test without color', () => {
    const tree = renderer
      .create(<CardLabel labelCaption="In Progress" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
