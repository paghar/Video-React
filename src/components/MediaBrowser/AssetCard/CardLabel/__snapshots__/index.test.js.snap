// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssetCardLabel /> AssetCardLabel Snapshot test 1`] = `
.c1 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  height: 27px;
  background-color: rgba(0,0,0,0.7);
  border-radius: 5px;
  text-align: center;
  padding-left: 19px;
  padding-top: 3px;
  padding-right: 10px;
}

.c2:before {
  content: ' ';
  width: 7px;
  height: 7px;
  background-color: green;
  border-radius: 100%;
  position: absolute;
  top: 10px;
  left: 8px;
}

<div
  className="c0"
>
  <span
    className="c1 c2"
  >
    In Progress
  </span>
</div>
`;

exports[`<AssetCardLabel /> AssetCardLabel Snapshot test without color 1`] = `
.c1 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  height: 27px;
  background-color: rgba(0,0,0,0.7);
  border-radius: 5px;
  text-align: center;
  padding-left: 8px;
  padding-top: 3px;
  padding-right: 7px;
}

<div
  className="c0"
>
  <span
    className="c1 "
  >
    In Progress
  </span>
</div>
`;
